src/riscv/lib/src/pvm/linux/parameters.rs

```diff
@@ -77,16 +79,155 @@ impl TryFrom<u64> for MainThreadId {
+/// Hard limit on the number of file descriptors that a system call can work with
+///
+/// We also use this constant to implictly limit how much memory can be associated with a system
+/// call. For example, `ppoll` takes a pointer to an array of `struct pollfd`. If we don't limit
+/// the length of that array, then we might read an arbitrary amount of memory. This impacts the
+/// proof size dramatically as everything read would also be in the proof.
+pub const RLIMIT_NOFILE: u32 = 512;
+
+/// Hard limit on the number of flock locks plus fcntl locks. Unused by us (and we only support one hart and no filesystem)
+const RLIMIT_LOCKS: u32 = 0;
+
+/// Hard limit on the number of signals for a user ID. Signals are not currently implemented.
+const RLIMIT_SIGPENDING: u32 = 0;
+
+/// Hard limit on the number of bytes that may be allocated for message queues. Message queues are not currently implemented.
+const RLIMIT_MSGQUEUE: u32 = 0;
+
+/// Hard limit on the niceness of a process. This is inverted and the actual niceness is 20 - the
+/// rlimit. We only support one hart
+const RLIMIT_NICE: u32 = 20;
+
+/// Hard limit on the real time priority of a process. We only support one hart.
+const RLIMIT_RTPRIO: u32 = 0;
+
+/// Hard limit on CPU time without making a blocking system call. We only support one hart.
+const RLIMIT_RTTIME: u32 = u32::MAX;
+
+pub struct Rlimit<MC: MemoryConfig>(pub u32, PhantomData<MC>);
+
+impl<MC: MemoryConfig> fmt::Debug for Rlimit<MC> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        self.0.fmt(f)
+    }
+}
+
+// Resource limits for system calls
+impl<MC: MemoryConfig> TryFrom<u64> for Rlimit<MC> {
+    type Error = Error;
+
+    fn try_from(value: u64) -> Result<Self, Self::Error> {
+        let value = value.try_into()?;
+
+        struct MemorySize<MC>(MC);
+        impl<MC: MemoryConfig> MemorySize<MC> {
+            const SIZE: u32 = MC::TOTAL_BYTES as u32;
+
+            const LOWER_BOUND: () = assert!(MC::TOTAL_BYTES >= u32::MIN as usize);
+            const UPPER_BOUND: () = assert!(MC::TOTAL_BYTES <= u32::MAX as usize);
+
+            // Hard limit on the size of the data segment
+            const RLIMIT_DATA: u32 = Self::SIZE;
+
+            // Hard limit on the size of the process stack
+            const RLIMIT_STACK: u32 = Self::SIZE;
+
+            // Hard limit on resident set size. Not used.
+            const RLIMIT_RSS: u32 = Self::SIZE;
+
+            // Hard limit on the memory that may be locked into RAM (B)
+            const RLIMIT_MEMLOCK: u32 = Self::SIZE;
+
+            // Hard limit on the size of a process's virtual memory (B)
+            const RLIMIT_AS: u32 = Self::SIZE;
+        }
+
+        let () = MemorySize::<MC>::LOWER_BOUND;
+        let () = MemorySize::<MC>::UPPER_BOUND;
+
+        match value {
+            0..=15 => {
+                let limit = match value {
+                    // Hard limit on CPU time (s)
+                    0 => RLIMIT_CPU,
```

Comment:
I feel like you could do away with the nested match statements and get rid of the `unreachable`, unless I'm missing a code path from your initial logic?

```suggestion
        let limit = match value {
            // Hard limit on CPU time (s)
            0 => RLIMIT_CPU,
            // other 1..15 cases
            _ => Err(Error::Search)
        };
        Ok(Rlimit::<MC>(limit, PhantomData)
```

Reply:
Done but the diff includes a rebase